x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/azureaistudio/AzureAiStudioServiceSettings.java

```diff
@@ -122,8 +120,8 @@ public String modelId() {
     @Override
     public void writeTo(StreamOutput out) throws IOException {
         out.writeString(target);
-        out.writeEnum(provider);
-        out.writeEnum(endpointType);
+        out.writeEnum(deploymentType);
```

Comment:
That's a bit tricky, the "old" Azure AI Studio integration is broken, but I think we still need to handle old ES nodes sending enum values of `AzureAiStudioProvider`, if we've a mixed cluster to at least not break on the transport level.

So we probably need to do something like this:

```
if out.getTransportVersion() < current new version:
   do not send deployment type and model, but the old fields
else:
   send the new fields deploymentType and model
```

Reply:
Speaking of transport versions, I think we need to introduce a new one in `TransportVersions`